docs/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,215 @@
+
+- `microsoft/dotnet:<version>-core` : that is **microsoft/dotnet:1.0.0-core**, image which runs [portable .NET Core applications](https://docs.microsoft.com/en-us/dotnet/articles/core/app-types) and it is optimized for running your application in **production**. It does not contain the SDK, and is meant to take the optimized output of `dotnet publish`. The portable runtime is well suited for docker container scenarios as running multiple containers benefit from shared image layers.  
+
+## Alternative images
+
+In addition to the optimized scenarios of development, build and production, we provide additional images:
+
+- `microsoft/dotnet:<version>-onbuild` : that is **microsoft/dotnet:onbuild**, contains [ONBUILD](https://docs.docker.com/engine/reference/builder/#/onbuild) triggers. The build will [COPY](https://docs.docker.com/engine/reference/builder/#/copy) you application, run `dotnet restore` and create an [ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#/entrypoint) `dotnet run` instruction to run the application when the Docker image is run. While not an optimized image for production, some may find it useful to simply copy their source code into an image and run it. 
+
+- `microsoft/dotnet:<version>-core-deps` : that is **microsoft/dotnet:1.0.0-core-deps**, if you wish to run self-contained applications use this image.  Contains the operating system with all of the native dependencies needed by .NET Core. Can also be used to build a custom copy of .NET core by compiling coreclr and corefx. While the **onbuild** variant is optimized to simply place your code in an image and run it, this image is optimized to have only the operating system dependencies required to run .NET Core apps that have the .NET Runtime packaged with the application. This image isn't generally optimized for running multiple .NET Core containers on the same host, as each image caries the .NET Core runtime within the application, and you will not benefit from image layering.   
+
+Latest versions of each variant:
+
+- `microsoft/dotnet` or `microsoft/dotnet:latest` (includes SDK)
+- `microsoft/dotnet:onbuild`
+- `microsoft/dotnet:core`
+- `microsoft/dotnet:core-deps`
+
+Here is a list of the images after a `docker pull <imagename>` on a development machine to show the various sizes. Notice, the development/build variant ('microsoft/dotnet:1.0.0-preview2-sdk) is larger as it contains the SDK to develop and build your application. 
+The production variant, `microsoft/dotnet:core` is smaller, as it only contains the .NET Core runtime. 
+The minimal image capable of being used on Linux, `core-deps`, is quite smaller, however your application will need to copy a private copy of the .NET Runtime with it. Since containers are already private isolation barriers, you will lose that optimization when running multiple dotnet based containers. 
```

Comment:
@BillWagner - Linux is called out here as it is a specific question commonly asked. What is the smallest image that can be used. 

What you are thinking (your second paragraph) is correct. It is a container made for optimized and isolated single purpose applications. 

@SteveLasker ^ and additional comments here?